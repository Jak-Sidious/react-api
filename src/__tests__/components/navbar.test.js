import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';

import Navigation from '../../components/Navbar/navbar';

Enzyme.configure({ adapter: new Adapter() });

describe('<Navigation />', () => {
  const component = mount(
    <MemoryRouter>
      <Navigation />
    </MemoryRouter>
  );
  it('should render without crashing', () => {
    const { enzymeWrapper } = mount(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    );
  });
  it('should render a <Menu> tag', () => {
    expect(component.find('Menu').length).toBe(2);
  });
  it('should render properly', () => {
    const wrapper = shallow(<Navigation />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should contain 3 Menu.Item', () => {
    const items = component.find('Menu.Item');
    expect(items).toHaveLength(0);
  });
});
