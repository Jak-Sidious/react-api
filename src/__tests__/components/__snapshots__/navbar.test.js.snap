// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render properly 1`] = `
<div>
  <Menu
    inverted={true}
    size="large"
  >
    <MenuItem
      active={false}
      as={[Function]}
      name="home"
      onClick={[Function]}
      to="/landing"
    />
    <MenuItem
      name="Categories"
    >
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        deburr={false}
        icon="dropdown"
        item={true}
        minCharacters={1}
        noResultsMessage="No results found."
        openOnFocus={true}
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        text="Categories"
        wrapSelection={true}
      >
        <DropdownMenu>
          <DropdownItem
            active={false}
            as={[Function]}
            name="Categories"
            onClick={[Function]}
            to="/catCreate"
          >
            CreateCategory
          </DropdownItem>
          <DropdownItem
            active={false}
            as={[Function]}
            name="Categories"
            to="/viewCat"
          >
            View Categories
          </DropdownItem>
        </DropdownMenu>
      </Dropdown>
    </MenuItem>
    <MenuItem>
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        closeOnBlur={true}
        deburr={false}
        icon="dropdown"
        item={true}
        minCharacters={1}
        noResultsMessage="No results found."
        onClick={[Function]}
        openOnFocus={true}
        renderLabel={[Function]}
        searchInput="text"
        selectOnBlur={true}
        selectOnNavigation={true}
        text="Recipes"
        wrapSelection={true}
      >
        <DropdownMenu>
          <DropdownItem>
            Create Recipe
          </DropdownItem>
          <DropdownItem>
            View Recipes
          </DropdownItem>
        </DropdownMenu>
      </Dropdown>
    </MenuItem>
    <MenuMenu
      position="right"
    >
      <Menu
        inverted={true}
      >
        <Dropdown
          additionLabel="Add "
          additionPosition="top"
          closeOnBlur={true}
          deburr={false}
          icon="dropdown"
          item={true}
          minCharacters={1}
          noResultsMessage="No results found."
          onClick={[Function]}
          openOnFocus={true}
          renderLabel={[Function]}
          searchInput="text"
          selectOnBlur={true}
          selectOnNavigation={true}
          wrapSelection={true}
        >
          <DropdownMenu>
            <DropdownItem
              name="Logout"
              onClick={[Function]}
            >
              Logout
            </DropdownItem>
          </DropdownMenu>
        </Dropdown>
      </Menu>
    </MenuMenu>
  </Menu>
</div>
`;
